examples/nlp/semantic_similarity_with_keras_nlp.py

```diff
@@ -0,0 +1,274 @@
+    dataset.drop(dataset[dataset["label"] == -1].index, inplace=True)
+
+    x = preprocessor((dataset["hypothesis"], dataset["premise"]))
+    y = np.array(dataset.label)
+    return x, y
+
+
+"""
+## BERT Baseline
+
+We'll use BERT from KerasNLP to establish a baseline.
+
+Initialize a BertPreprocessor object from KerasNLP library with preset 
+configuration "bert_tiny_en_uncased". The preprocessor can preprocess 
+sentence pairs for BERT-based models by converting tokens to corresponding IDs in 
+BERT vocabulary and adding special tokens.
+"""
+bert_preprocessor = keras_nlp.models.BertPreprocessor.from_preset(
+    "bert_tiny_en_uncased"
+)
+
+"""
+Preprocessing the sentence pairs in the training, validation, and test data using 
+a preprocessor object. BertPreprocessor automatically takes care of sentence packing,
+and seperates them with [SEP] token
+"""
+x_train, y_train = preprocess_dataset(snli_train, bert_preprocessor)
+x_test, y_test = preprocess_dataset(snli_test, bert_preprocessor)
+x_val, y_val = preprocess_dataset(snli_val, bert_preprocessor)
+
+"""
+### Train the Model End to End
+ 
+BertClassifier class attaches classification head to the BertBackbone backbone, mapping 
+the backbone outputs to logit output suitable for a classification task. This significantly
+reduces need of custom code.
+
+Here we'll use this model with pre-trained weights. `from_preset()` method allows you
+to use your own preprocessor. Here we'll set the `num_classes` as 3 for SNLI dataset
+"""
+
+bert_clf = keras_nlp.models.BertClassifier.from_preset(
+    "bert_tiny_en_uncased", num_classes=3, preprocessor=None
+)
+
+bert_clf.summary()
+
+"""
+KerasNLP models compile automatically. Let's train the model we just instantiated, using 
+the fit() method with the following arguments:
+
+    x_train: The input data for the training set, preprocessed using BertPreprocessor
+    y_train: The target labels for the training set
+    validation_data: A tuple containing the input and target data for the validation set
+    epochs: The number of epochs to train the model
+    batch_size: The number of samples per gradient update
+"""
+
+bert_clf.fit(
+    x=x_train, y=y_train, validation_data=(x_val, y_val), epochs=1, batch_size=512
+)
+
+"""
+
+Our BERT classifier gave us ~68% accuracy on validation split, let's see how it performs
+on Test split
+
+### Evaluate the performance of the trained model on test data.
+"""
+bert_clf.evaluate(x=x_test, y=y_test)
+
+"""
+Our baseline bert gave almost similar (~68%) accuracy on the test split. Let's see if we can
+improve it.
+
+Let's recompile our model with a different learning rate and see performance
+"""
+bert_clf_1 = keras_nlp.models.BertClassifier.from_preset(
+    "bert_tiny_en_uncased", num_classes=3, preprocessor=None
+)
+
+bert_clf_1.compile(
+    loss=keras.losses.SparseCategoricalCrossentropy(from_logits=True),
+    optimizer=tf.keras.optimizers.Adam(1e-4),
```

Comment:
Maybe show off AdamW here? That should boost performance somewhat.

Also you could try showing a learning rate with warmup and linear decay. I'm not sure that will show too much different with the "tiny" checkpoints, but with larger ones it should.

Here's a simple learning rate schedule I have used in the past.

```

class TriangularSchedule(keras.optimizers.schedules.LearningRateSchedule):
    """Linear ramp up for `warmup` steps, then linear decay to zero at `total` steps."""
    def __init__(self, rate, warmup, total):
        self.rate = tf.cast(rate, dtype="float32")
        self.warmup = tf.cast(warmup, dtype="float32")
        self.total = tf.cast(total, dtype="float32")

    def __call__(self, step):
        step = tf.cast(step, dtype="float32")
        multiplier = tf.cond(
            step < self.warmup,
            lambda: step / self.warmup,
            lambda: (self.total - step) / (self.total - self.warmup),
        )
        return tf.maximum(self.rate * multiplier, 0.0)
```

This might be useful to demo, as it's a very common learning rate schedule in NLP.

Reply:
Thanks for the input, with bert tiny x triangular rate scheduler, val and test accuracy hikes up to 85%

Reply:
Also AdamW on 1st epoch with a LR scheduler gives 65% of val accuracy, way lesser than adam, moving with Adam only here, what say @mattdangerw ?